Add Tekton Task and TaskRun to create and delete GitHub webhooks

a-roberts opened the pull request:
Tested this with github.com and github.ibm.com using PATs and secrets created through the Tekton Dashboard's Webhook Extension.

This is for https://github.com/tektoncd/experimental/issues/241

# Changes

Credits to @akihikokuroda for the create webhook work this is largely based on 😄 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

a-roberts commented:
> Looks good to me! One question: How can the user find the webhook name (id) except from the url of each "Manage Webhook" page?

I think that's the only way at the moment - so if I've understood your point correctly, as you know the way we have it now (current UI code) is different in that we use a  named GitHubSource and  a ConfigMap and that's where we retrieve the webhook "names" from in our table. 

This is unless we were to also offer a "get webhooks for repository URL page" perhaps and surface that eventually through the UI, but I'm sure we'll be sticking with a ConfigMap so we'll let users create "named" hooks instead of using IDs. 

The create/delete code here in the yaml would remain the same though as the name has to be `web` on the POST and you delete by the ID.

a-roberts commented:
@dibbles could I get a look at this as well please? Aki's given us the thumbs up too so wanting to be sure this is the direction we want to run with 😸

dibbles commented on webhooks-extension/config/triggers_prototype/github/README.md:
```diff
@@ -0,0 +1,18 @@
+- Apply the Task definition:
+`kubectl apply -f config/triggers_prototype/github/webhook.yaml`
+
+- Create a GitHub secret, for example with the Tekton Dashboard or importing it using the Tekton Webhooks Extension. You will refer to its name in the next step.
```
I dont think we should advocate importing secrets from a git repo.  Also I would saying "using the Tekton Dashboard" rather than "with"

dibbles commented on webhooks-extension/config/triggers_prototype/github/webhook.yaml:
```diff
@@ -0,0 +1,93 @@
+      description: Whether or not to create or delete the webhook
+      default: "create"
+    - name: ExternalUrl
+      description: The external access URL (used for creating the webhook)
```
make this clearer i.e for example the ingress external URL in front of the EventListener

dibbles commented on webhooks-extension/config/triggers_prototype/github/webhook.yaml:
```diff
@@ -0,0 +1,93 @@
+    - name: ExternalUrl
+      description: The external access URL (used for creating the webhook)
+      default: "replacemeip.nip.io"
+    - name: GithubOwner 
```
I wonder if this is GitHubOwnerOrOrgName rather than the Owner - could just change the description though??  Note - has this been tested against an org?

Also weird use of camel casing GitHub in descriptions but not on parameter names

a-roberts replied:
Replaced param names, it's owner yeah

dibbles commented on webhooks-extension/config/triggers_prototype/github/webhook.yaml:
```diff
@@ -0,0 +1,93 @@
+      default: "1234567"
+  steps:  
+  - name: create-or-delete-webhook
+    image: pstauffer/curl:latest   
```
have image licenses been checked?

a-roberts replied:
Looking at using curlimages/curl:latest as it looks to be official, see https://github.com/curl/curl-docker and note it's under the verified curl org

dibbles commented:
Closing as superceded by https://github.com/tektoncd/experimental/pull/276

dibbles commented:
> Closing as superceded by #276

whoops - closed the wrong one!

a-roberts commented:
@dibbles I've rebased on master and used an official curl image, replaced some param names so they're GitHub not Github, and tested I get a working webhook with my event listener using the latest version of Triggers (readme updated/merged in with the Ingress docs too)

dibbles commented:
/approve
/lgtm
/meow boxes
